dusty/compiler/spec_assembler.py

```diff
@@ -105,10 +104,10 @@ def get_assembled_specs():
     return specs
 
 def get_specs_repo():
-    return get_config_value(constants.CONFIG_SPECS_REPO_KEY)
+    return Repo(get_config_value(constants.CONFIG_SPECS_REPO_KEY))
 
 def get_specs_path():
-    return local_repo_path(get_specs_repo())
+    return get_specs_repo().local_path
```

Comment:
if you move these two functions to path.py (arguable one of them should live there anyway) and get_all_repos.py to source.py.  The circular import goes away.  resolve and override then become easier/simpler